robots/pkg/kubevirt/cmd/remove/jobs_test.go

```diff
@@ -0,0 +1,555 @@
+	}
+	tests := []struct {
+		name string
+		args args
+		wantModification bool
+	}{
+		{
+			name: "should modify",
+			args: args{
+				releases: []*github.RepositoryRelease{
+					newRelease("v1.22.0"),
+					newRelease("v1.21.0"),
+					newRelease("v1.20.0"),
+					newRelease("v1.19.0"),
+				},
+				removeJobsOpts: removeJobsOptions{
+					jobConfigPathKubevirtPeriodics: "testdata/should_modify/kubevirt-periodics.yaml",
+					jobConfigPathKubevirtPresubmits: "testdata/should_modify/kubevirt-presubmits.yaml",
+				},
+			},
+			wantModification: true,
+		},
+		{
+			name: "should not modify",
+			args: args{
+				releases: []*github.RepositoryRelease{
+					newRelease("v1.22.0"),
+					newRelease("v1.21.0"),
+					newRelease("v1.20.0"),
+					newRelease("v1.19.0"),
+				},
+				removeJobsOpts: removeJobsOptions{
+					jobConfigPathKubevirtPeriodics: "testdata/should_not_modify/kubevirt-periodics.yaml",
+					jobConfigPathKubevirtPresubmits: "testdata/should_not_modify/kubevirt-presubmits.yaml",
+				},
+			},
+			wantModification: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempDir, err := os.MkdirTemp("", "removeOldJobsIfNewOnesExist-")
+			checkErr(err)
+			defer os.RemoveAll(tempDir)
+
+			copyFiles([]string{tt.args.removeJobsOpts.jobConfigPathKubevirtPeriodics, tt.args.removeJobsOpts.jobConfigPathKubevirtPresubmits}, tempDir)
+			removeJobsOpts = removeJobsOptions{
+				jobConfigPathKubevirtPeriodics: filepath.Join(tempDir, filepath.Base(tt.args.removeJobsOpts.jobConfigPathKubevirtPeriodics)),
+				jobConfigPathKubevirtPresubmits: filepath.Join(tempDir, filepath.Base(tt.args.removeJobsOpts.jobConfigPathKubevirtPresubmits)),
+			}
+			removeOldJobsIfNewOnesExist(tt.args.releases)
+
+			sum256OrigPeriodics, sum256TempCopyPeriodics := hashFiles(tt.args.removeJobsOpts.jobConfigPathKubevirtPeriodics, removeJobsOpts.jobConfigPathKubevirtPeriodics)
+			sum256OrigPresubmits, sum256TempCopyPresubmits := hashFiles(tt.args.removeJobsOpts.jobConfigPathKubevirtPresubmits, removeJobsOpts.jobConfigPathKubevirtPresubmits)
+
+			if tt.wantModification {
+				if reflect.DeepEqual(sum256OrigPeriodics, sum256TempCopyPeriodics) {
+					t.Errorf("removeOldJobsIfNewOnesExist(), wantModification %v, got %v", tt.wantModification, false)
+				}
+				if reflect.DeepEqual(sum256OrigPresubmits, sum256TempCopyPresubmits) {
+					t.Errorf("removeOldJobsIfNewOnesExist(), wantModification %v, got %v", tt.wantModification, false)
+				}
+			} else {
+				if !reflect.DeepEqual(sum256OrigPeriodics, sum256TempCopyPeriodics) {
+					t.Errorf("removeOldJobsIfNewOnesExist(), wantModification %v, got %v", tt.wantModification, true)
+				}
+				if !reflect.DeepEqual(sum256OrigPresubmits, sum256TempCopyPresubmits) {
+					t.Errorf("removeOldJobsIfNewOnesExist(), wantModification %v, got %v", tt.wantModification, true)
+				}
+			}
+		})
+	}
+}
+
+func hashFiles(original string, tempCopy string) ([32]byte, [32]byte) {
```

Comment:
hmm this function has duplication, why don't we have just `hashFile` that takes a file path and returns its `sha256.Sum256`?